src/services/strike_service.py

```diff
@@ -13,22 +21,121 @@ async def add_strike(
+    author: discord.Member,
 ):
     # find user in DB
     db_user = users_database.get_user(user.id)
     if db_user is None:
         log_info_and_embed(
             logging_embed,
             logger,
-            f"User not found in database, creating new record",
+            "User not found in database, creating new record",
         )
-        db_user_id = users_database.add_user(user.id)
-        logger.info(f"Created user record in DB with id {db_user_id}")
-
-        db_user = create_empty_user(db_user_id, user.id, None, None)
+        db_user = users_database.add_user(user.id)
 
     # create strike
+    strike_timestamp = datetime.now()
+    strike = Strike(
+        user_id=db_user.user_id,
+        severity=severity,
+        reason=reason,
+        created_timestamp=strike_timestamp,
+        created_by=author.id,
+        last_edited_timestamp=strike_timestamp,
+        last_edited_by=author.id,
+    )
+    strike.strike_id = strikes_database.add_strike(strike)
+    logging_embed.set_footer(text=f"Strike ID: {strike.strike_id}")
 
     # increment user points, update
+    db_user.last_infraction_timestamp = strike_timestamp
+    previous_points = db_user.get_strike_points()
+    _apply_strike_point_penalty(db_user, severity)
+    users_database.update_user_strike_points(db_user)
+
+    log_info_and_embed(
+        logging_embed,
+        logger,
+        f"<@{user.id}> was given a strike, bringing them to {db_user.get_strike_points()} points",
+    )
+
+    punishment = await _apply_punishment(logging_embed, user, db_user, previous_points)
+    logging_embed.add_field(name="Punishment", value=punishment)
+
+    # message user
+    try:
+        await send_dm(
+            user,
+            f"Your actions in NA Ultimate Raiding - FFXIV resulted in a strike against your account. This may result in punishment depending on the frequency or severity of your strikes.\n**Reason:** {reason}",
+        )
+    except Exception as e:
+        log_info_and_embed(
+            logging_embed, logger, f"Failed to send DM to exiled user, {e}"
+        )
+
+
+def _apply_strike_point_penalty(db_user: User, severity: StrikeSeverity):
+    match severity:
+        case StrikeSeverity.MINOR:
+            db_user.temporary_points = (
+                db_user.temporary_points + MINOR_INFRACTION_POINTS
+            )
+        case StrikeSeverity.MODERATE:
+            db_user.temporary_points = (
+                db_user.temporary_points + MODERATE_INFRACTION_POINTS
+            )
+        case StrikeSeverity.SERIOUS:
+            db_user.permanent_points = (
+                db_user.permanent_points + SERIOUS_INFRACTION_POINTS
+            )
+
+
+async def _apply_punishment(
+    logging_embed: discord.Embed,
+    user: discord.Member,
+    db_user: User,
+    previous_points: int,
+) -> str:
+    total_points = db_user.get_strike_points()
+
+    # TODO: known error, if an exiled user is given a strike, the follow up exile is not created
```

Comment:
if we're not going to handle this now, let's attach a ticket to this

Reply:
Ticket added MOD-93 also called out the ticket number in this todo